agent/iskakalunan.py

```diff
@@ -0,0 +1,382 @@
+        if job_empty:
+            messages = self.fetch_messages(LABEL_IN_QUEUE)
+            logger.info("Check In Queue tasks: %d tasks in queue" % len(messages))
+            if messages:
+                update_body = {"removeLabelIds": [LABEL_IN_QUEUE],
+                               "addLabelIds": [LABEL_IN_PROCESS]}
+                self.updateLabels(messages[0], update_body)
+                self.reply_status(messages[0], MSG_TYPE_UNDER_EXECUTION)
+                self.prepare_job(messages[0])
+
+    def fetch_messages(self, label_id):
+        """Fetch messages from pre-set mail account
+
+        valid results will be updated to messages
+            - Valid result means messages contain 'Iskakalunan' (case insensitive)
+            - Only messages tags w/ 'NewRequest', 'InQueue', and 'InProcess'
+        """
+        try:
+            response = self.service.users().messages().list(userId='me', labelIds=label_id).execute()
+            message_list = []
+            messages = []
+            if 'messages' in response:
+                message_list.extend(response['messages'])
+
+            while 'nextPageToken' in response:
+                page_token = response['nextPageToken']
+                response = service.users().messages().list(userId=user_id,
+                                                           labelIds=label_id,
+                                                           pageToken=page_token).execute()
+                message_list.extend(response['messages'])
+
+            for message in message_list:
+                message_content = self.service.users().messages().get(userId='me', id=message['id']).execute()
+                message_temp = Message(message['id'], message['threadId'], message_content)
+                if 'moztpeqa' not in message_temp.sender:
+                    logger.debug("message sender: %s" % message_temp.sender)
+                    messages.append(message_temp)
+
+            return messages
+        except errors.HttpError, error:
+            logger.error("An error occurred: %s" % error)
+
+    def update_labels(self, msg, update_labels):
+        try:
+            logger.debug('Update labels for msgId: %s with %s' % (msg.message_id, update_labels))
+            self.service.users().messages().modify(userId='me', id=msg.message_id, body=update_labels).execute()
+        except errors.HttpError, error:
+            logger.error('An error occurred: %s' % error)
+
+    def prepare_try_build(self, revision):
+        try:
+            url_try_builds = "https://archive.mozilla.org/pub/firefox/try-builds/"
+            req = urllib2.Request(url_try_builds)
+            response = urllib2.urlopen(req)
+            html_try_builds = response.read()
+
+            pattern = 'href="(.+' + message.revision + '\/)"'
+            m = re.search(pattern, html_try_builds)
+            url_revision_build = "https://archive.mozilla.org" + m.group(1) + "try-win64/firefox-55.0a1.en-US.win64.zip"
```

Comment:
Currently, the first release is targeted at Windows 64-bit. There should be a separate patch for migrating the code to support all platforms in the future.